doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc

```diff
@@ -0,0 +1,96 @@
+[id='con-traits_{context}']
+
+= Traits in the {DECISION_ENGINE}
+Traits in {DECISION_ENGINE} allow you to model multiple dynamic types that do not fit in a class hierarchy. A trait is an interface that you can apply (and eventually remove) to an individual object at runtime. However, traits are not supported in the executable rule model, therefore traits functionality and its specific implementation code is moved to an individual module named `drools-traits`. To use traits, you can import the `drools-traits` module to Maven.
+
+The `drools-traits` module is based on the `drools-core` and `drools-compiler` module. The classes from `drools-core` and `drools-compiler` are moved to the `drools-traits` module, containing `.traits` prefix in package name and `-Impl` suffix in class name to avoid the split package problem.
+
+NOTE: A class or an interface that is used specifically by traits is moved to the `drools-traits` module.
+
+== Components in traits
+
+Traits require the following components:
+
+[source]
+----
+org.drools.core.reteoo.KieComponentFactoryFactory=org.drools.traits.core.reteoo.TraitKieComponentFactoryFactory
+org.drools.compiler.builder.impl.TypeDeclarationBuilderFactory=org.drools.traits.compiler.builder.impl.TraitTypeDeclarationBuilderFactory
+----
+
+To use traits, you must instantiate the specific traits classes using the service loader mechanism. For example, the `kie.conf` file in `drools-traits` module supports the execution of traits related tests that are available in the `drools-core` and `drools-compiler` module:
+
+.Example objects in`kie.conf`
+
+[source]
+----
+org.drools.core.reteoo.KieComponentFactoryFactory=org.drools.traits.core.reteoo.TraitKieComponentFactoryFactory
+org.drools.compiler.builder.impl.TypeDeclarationBuilderFactory=org.drools.traits.compiler.builder.impl.TraitTypeDeclarationBuilderFactory
+org.kie.api.persistence.jpa.KieStoreServices = org.drools.persistence.jpa.KnowledgeStoreServiceImpl
+?org.kie.internal.process.CorrelationKeyFactory = org.jbpm.persistence.correlation.JPACorrelationKeyFactory
+----
+
+IMPORTANT: Persistence related tests are moved to the `drools-traits` module. Therefore, the `kie.conf` file contains `KieStoreServices` defined for persistence.
```

Comment:
This is also irrelevant, up until the end I'd say.

Reply:
@lucamolteni  Would you suggest keeping only the introduction paragraph and removing the rest content including components?

Reply:
yes @hmanwani-rh

Reply:
@lucamolteni Agreed on everything.